apiservers/engine/backends/container.go

```diff
@@ -173,22 +174,39 @@ func (c *Container) ContainerCreate(config types.ContainerCreateConfig) (types.C
 	}
 
 	plCreateParams := c.dockerContainerCreateParamsToPortlayer(config, layer.ID, host)
-	createResults, err := client.Exec.ContainerCreate(plCreateParams)
-
+	createResults, err := client.Containers.Create(plCreateParams)
 	// transfer port layer swagger based response to Docker backend data structs and return to the REST front-end
 	if err != nil {
-		if _, isa := err.(*exec.ContainerCreateNotFound); isa {
+		if _, isa := err.(*containers.CreateNotFound); isa {
 			return types.ContainerCreateResponse{}, derr.NewRequestNotFoundError(fmt.Errorf("No such image: %s", layer.ID))
 		}
 
 		// If we get here, most likely something went wrong with the port layer API server
-		return types.ContainerCreateResponse{},
-			derr.NewErrorWithStatusCode(fmt.Errorf("Unknown error from the exec port layer"), http.StatusInternalServerError)
+		return types.ContainerCreateResponse{}, derr.NewErrorWithStatusCode(err, http.StatusInternalServerError)
+	}
+
+	id := createResults.Payload.ID
+	h := createResults.Payload.Handle
+
+	// configure networking
+	netConf := toModelsNetworkConfig(config)
+	if netConf != nil {
+		_, err := client.Scopes.AddContainer(
```

Comment:
For the handle model to work, we should be using the handle returned by AddContainer to pass into commit.